Scripts/RemediationScripts/Remediate-NSGConfigurationOnApplicationGatewaySubnet.ps1

```diff
@@ -0,0 +1,856 @@
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "[Step 3 of 4] Back up Subnet(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        # Backing up Subnet(s) details.
+        $backupFile = "$($backupFolderPath)\SubnetDetailsBackUp.csv"
+        $SubnetWithoutNSGConfigured | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "Subnet(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else
+    {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    if (-not $DryRun)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enable the DDoS Protection Plan on Subnet(s) in the Subscription..." 
+        Write-Host $([Constants]::SingleDashLine)
+        
+
+        if (-not $Force)
+        {
+            Write-Host "Do you want to configure NSG on the Subnet(s) in the Subscription? " -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "we are starting the procedure to configure the NSG on the Subnet(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+        }
+      
+
+        # List for storing remediated Subnet(s)
+        $SubnetRemediated = @()
+
+        # List for storing skipped Subnet(s)
+        $SubnetSkipped = @()
+
+        Write-Host "Enabling the NSG on Subnet(s)..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Loop through the list of Subnet(s) which needs to be remediated.
+        $SubnetWithoutNSGConfigured | ForEach-Object {
+            $subnet = $_
+            try
+            {
+                
+                    Write-Host "To Start configuring the NSG on the Subnet(s), Please enter the Network Security Group Name..." -ForegroundColor $([Constants]::MessageType.Info)
+                    $NSGName = Read-Host -Prompt "Please enter name of Network Security Group"
+                    $NSGRGName = Read-Host -Prompt "Please enter Resource Group of Network Security Group"
+                    if($NSGName -ne $null -and $NSGRGName -ne $null)
+                    {
+                        $nsg = Get-AzNetworkSecurityGroup -ResourceGroupName $NSGRGName -Name $NSGName
+                        if($NSGName -ne $null)
```

Comment:
Should this check be performed on '$nsg'?

Reply:
yes, because User will provide the NSG name only and we need to get the NSG object with the Given name, if that does not exist, we would not be able to remediate.